[wasm][test-browser] Retry launching chrome if it fails on startup

radical opened the pull request:
When starting chrome via selenium+chromedriver, it seems to crash
inexplicably (https://github.com/dotnet/runtime/issues/43955).
As a workaround, let's retry that a few times.

Also, adds `--no-sandbox` from https://github.com/dotnet/xharness/pull/340 .

radical commented:
/cc @lewing @akoeplinger

akoeplinger commented:
I don't think we should be adding `--no-sandbox` as the docs say:

```
A common cause for Chrome to crash during startup is running Chrome as root user (administrator) on Linux. While it is possible to work around this issue by passing '--no-sandbox' flag when creating your WebDriver session, such a configuration is unsupported and highly discouraged. Please configure your environment to run Chrome as a regular user instead.
```

We're definitely not running Chrome as root so I'm a bit hesitant to add this highly discouraged option. The retry should hopefully fix it too.

akoeplinger commented on src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs:
```diff
@@ -87,7 +88,29 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
+                {
+                    return (driverService, new ChromeDriver(driverService, options, _arguments.Timeout));
+                }
+                catch (WebDriverException wde) when (wde.Message.Contains("exited abnormally") && retry_num < max_retries - 1)
```
should we make this check for `failed to start: exited abnormally` so we don't accidentally retry other cases of abnormal exits, e.g. if the browser crashes due to some WASM bug while running tests.

radical replied:
This is catching exceptions only while starting the browser, before any page gets opened.

akoeplinger commented:
We could also try passing the options puppeteer passes by default: https://github.com/puppeteer/puppeteer/blob/main/src/node/Launcher.ts#L159-L181

radical commented:
I have removed `--no-sandbox`, and added some extra arguments for chrome based on @akoeplinger's comment. I removed the ones that seem to be passed already (by chromedriver/selenium?).
